src/graph.py

```diff
@@ -752,6 +741,105 @@ def perform_GO_enrichment_on_clusters(nxgraph, clustering):
     """
     pass
 
+
+###############################################################################
+            #######            Results           #######
+###############################################################################
+
+def summarize_grid_search(results, mode, top_n=False): 
+    """
+    Summarizes results of `_grid_randomization` into a matrix where each row is a gene 
+    and each column is a parameter run. If summarizing "membership", entries will be 0 or 1
+    indicating whether or not a node appeared in each experiment. If summarizing "robustness"
+    or "specificity", entries indicate robustness or specificity values for each experiment. 
+    Also, node attributes columns are added for plotting purposes.
+
+    Arguments:
+        results (list of tuples): Results of `_grid_randomization` (paramstring, forest, augmented forest)
+        mode (str): Reported values "membership", "robustness", "specificity"
+        top_n (int): Takes the top_n values of the summary dataframe. top_n=-1 sets no threshold
+
+    Returns:
+        pd.DataFrame: Columns correspond to each parameter experiment, indexed by nodes
+    """
+    
+    # Exclude any degenerate results
+    results = {paramstring: graphs for paramstring, graphs in results.items() if graphs["augmented_forest"].number_of_nodes() > 0}
+    
+    if mode == "membership": # Summarize single-run parameter search
+        series = [pd.Series(1, index=graphs["augmented_forest"].nodes(), name=paramstring) for paramstring, graphs in results.items()]
+    elif mode == "robustness": # Summarize randomized robustness
+        series = [get_networkx_graph_as_dataframe_of_nodes(graphs["augmented_forest"])["robustness"].rename(paramstring) for paramstring, graphs in results.items()]
+    elif mode == "specificity": # Summarize randomized specificity
+        series = [get_networkx_graph_as_dataframe_of_nodes(graphs["augmented_forest"])["specificity"].rename(paramstring) for paramstring, graphs in results.items()]
+    else:
+        logger.info("`mode` must be one of the following: 'membership', 'robustness', or 'specificity'.")
```

Comment:
This is more than just info. Call it warning or critical. 

Check out all the methods here: https://docs.python.org/3/library/logging.html#logging.Logger.info

Reply:
401ae1054a10beddf2099b7d94bf990c91a602bc